### ISSUE TITLE
Tomas Salgado

### ISSUE THREAD

**Issue by mn476** [cucapra/undergrad-research#97]
**Note.** Before applying, make sure you can receive notifications from Github. Once you apply, make sure you're subsribed to receive emails when someone responds to the issue. We will respond to your application on Github first so it's important that you receive notifications to it and respond quickly.

We are thrilled that you're interested in research in our research group! Please fill out this issue template. When you submit it, we will get a notification in our group chat. Please include any relevant details you can think of! Here are [examples](https://github.com/cucapra/undergrad-research/issues?q=is%3Aissue+is%3Aopen+label%3AExample) of filled out templates for reference.

For active projects in our group, take a look at our [group website](https://capra.cs.cornell.edu/).

## Personal Details

**Name**: Tomas Salgado

**Major**: Computer Science and Mathematics

**Year in Cornell & Expected graduation date:** Sophomore, Expected Graduation: May 2025

**Relevant classes**: CS 2110, CS 2800, CS 1110, Linear Algebra, Real Analysis

**Interested in continuing research during the summer?**: Yes

**Expertise (languages/frameworks/etc.):**: Python, Java, have worked with Tensorflow

## Research

**When do you want to do research?**
(Researchers can get involved during the semester as an "extracurricular", equivalent to a 3-4 credit class, or get more involved over the summer as a full-time job.)

I would like to do research during this academic school year. 

**What is exciting to you about research?**
(How did you get interested in it? What are you hoping to get out it?)

I love learning, and I love working on hard problems. This is what research is all about. I particularly like how research differs from classes, which are more structured, but also give less room to explore and learn new things. I'm hoping that with this research group I will be able to improve my technical skills and get the chance to work on difficult problems. 

**What kind of research do you want to do?**
(It's OK to say, "I don't know; I'm looking to explore!")
I am particularly interested in research related to programming languages, machine learning, and natural language processing, but I am also very open to exploring more options. 

## Background

**Note**: While these questions are optional for first & second year students, we highly encourage everyone to respond to them. Third & fourth year students are required to respond to all questions.

**Was there a paper that particularly excited you?**
(This doesn't have to be a paper from our group.)
"Reticle: A Virtual Machine for Programming Modern FPGAs"
I found it very interesting how Reticle was able to significantly speed up compilation times. I have always found it fascinating how almost any process can be accelerated by creating a new, carefully-constructed machine or language.  

**Which of the [current research projects](https://capra.cs.cornell.edu/) would you be interested in working on and why?**

Braid, a Safe Heterogeneous Language for Real-Time Graphics
Braid catches my attention because to me the concept of heterogenous programming is fascinating. Additionally, I got some exposure to the field of Computer Graphics in my research last semester, and so I find it exciting to learn about a language that can assist in the generation of computer graphics. 

Dahlia, a Language for Predictable Accelerator Design
I find it fascinating how this language is focused on making certain concepts related to HLS programs more efficient than any other modern languages. 

Exploiting Temporal Redundancy for Live Computer Vision
This project interests me because it uses neural networks in computer vision, which combines two fields that seem particularly engaging and full of potential these days. 

**Anything else you want to tell us about yourself?**

**Attach a CV/Resumé**
[Resume.pdf](https://github.com/cucapra/undergrad-research/files/9901398/Resume.pdf)
